src/tests/TestFile.cpp

```diff
@@ -14,28 +14,51 @@ namespace tester {
 
 uint64_t TestFile::nextId = 0;
 
-TestFile::TestFile(const fs::path& path) : id(++nextId), testPath(path) {
+TestFile::TestFile(const fs::path& path, const fs::path& tmpPath)
+  : testPath(path) {
 
-  // create a unique temporary file to use as the inputs stream path
-  std::string fileInsPath = stripFileExtension(testPath.filename());
-  insPath = fs::temp_directory_path() / (fileInsPath + std::to_string(id) + ".ins");
-  outPath = fs::temp_directory_path() / (fileInsPath + std::to_string(id) + ".out");
+  fs::path testDir = tmpPath / std::to_string(nextId);
+  setInsPath(testDir / "test.ins");
+  setOutPath(testDir / "test.out");
 
-  std::ofstream makeInsFile(insPath);
-  std::ofstream makeOutFile(outPath);
+  try {
+    // Create tmp directory if it doesn't exist
+    std::cout << "Attempting to create directory: " << testDir << std::endl;
+    if (!fs::exists(testDir)) {
+      if (!fs::create_directories(testDir)) {
+        throw std::runtime_error("Failed to create directory: " + testDir.string());
+      }
+    }
+    // Create the temporary input and ouput files
+    std::ofstream createInsFile(insPath);
```

Comment:
I don't think there's any reason to create the files here. Opening and immediately closing the files is just extra file system interaction. The calls to `open` have `O_CREAT` on them, so they should make new files. Making the directory is important though.

Reply:
Thanks good point, I will create the files when the need to be used in that case.